s3datarepository/iam.go

```diff
@@ -241,132 +422,292 @@ func (s *S3Repository) GrantAccess(ctx context.Context, id string, derivative bo
 		}()
 	})
 
-	log.Debugf("created role %s", roleName)
+	var instanceRoleAssociated bool
 
-	// create instance profile
-	instanceProfileOutput, err := s.IAM.CreateInstanceProfileWithContext(ctx, &iam.CreateInstanceProfileInput{
-		InstanceProfileName: aws.String(roleName),
-		Path:                aws.String(s.IAMPathPrefix),
-	})
-	if err != nil {
-		return nil, ErrCode("failed to create instance profile "+roleName, err)
-	}
+	// if the instance already has some other instance profile, we copy all policies
+	// from the existing profile into the new one and then disassociate the old profile
+	if instanceInfo.IamInstanceProfile != nil {
+		// we only have the instance profile arn, so let's extract the name
+		ipArns := strings.Split(aws.StringValue(instanceInfo.IamInstanceProfile.Arn), "/")
+		currentInstanceProfileName := ipArns[len(ipArns)-1]
+		if currentInstanceProfileName == roleName {
+			instanceRoleAssociated = true
+		}
```

Comment:
this appears to be a useless variable assignment?  just do that in the if block.

Reply:
it used further down to determine whether we need to associate the instance profile